### PR TITLE
Address issue 1004 ("\n in mo.md literally interperted in markdown mode")

### PR DESCRIPTION
**Pull request by fuenfundachtzig**
This only replaces the `dedent` library by `dedent-js`.

For details, see https://github.com/marimo-team/marimo/issues/1004.

### CODE REVIEW

**Comment by dmadisetti** [comment]
dedent-js hasn't been maintained in 10yrs, I think dendent is the correct dependency

Maybe just open an issue in dedent? Alternatively, it's a trivial dep and could be moved to just utils with the patch

**Comment by akshayka** [comment]
> dedent-js hasn't been maintained in 10yrs, I think dendent is the correct dependency
> 
> 
> 
> Maybe just open an issue in dedent? Alternatively, it's a trivial dep and could be moved to just utils with the patch

I wasn't aware it wasn't maintained.

I'd be fine with either of those solutions!

**Comment by dmadisetti** [comment]
I think this package works: https://github.com/jridgewell/string-dedent

It's listed as the example in the proposal to add String.dedent to JavaScript, so would probably get polyfill support too if the proposal goes through

**Comment by akshayka** [comment]
@fuenfundachtzig would you like to update this PR to use [string-dedent](https://github.com/jridgewell/string-dedent) instead?

**Comment by fuenfundachtzig** [comment]
Sure, can do!

**Comment by fuenfundachtzig** [comment]
Now using [string-dedent](https://github.com/jridgewell/string-dedent).

File: frontend/src/core/codemirror/language/__tests__/markdown.test.ts [hunk 1]
```diff
@@ -119,22 +119,22 @@ describe("MarkdownLanguageAdapter", () => {
       const pythonCode = 'mo.md("""   \n# Title\nContent\n   """)';
       const [innerCode, offset] = adapter.transformIn(pythonCode);
       expect(innerCode).toBe("# Title\nContent");
-      expect(offset).toBe(13);
+      expect(offset).toBe(9);
     });
 
-    it("should handle space around the f=strings", () => {
+    it("should handle space around the f-strings", () => {
```
Comment (dmadisetti) [inline frontend/src/core/codemirror/language/__tests__/markdown.test.ts:125]: it("doesn't do this")

Looks like a quick copypaste from a different test, will update in my pr

**Review by dmadisetti — COMMENTED** [review]
LGTM, I added some tests in https://github.com/fuenfundachtzig/marimo/pull/1
and they pass for me.

**Review by akshayka — APPROVED** [review]
Woohoo! Thank you @fuenfundachtzig and @dmadisetti.

**Pull request merged**